scripts/install-protobuf.sh

```diff
@@ -16,17 +17,24 @@ echo "fetching ${protoc_url}"
 curl -L -o "protoc.zip" ${protoc_url}
 
 # Unzip
+rm -rf protoc3
 echo "extracting..."
 unzip -u protoc.zip -d protoc3
 
-
 # create local devtools dir.
+rm -rf devtools
 mkdir devtools
 mkdir devtools/bin
 mkdir devtools/include
 
-echo "moving bin/protoc to ./devtools/bin/protoc"
-mv protoc3/bin/* devtools/bin/
+# Don't reinstall protoc if it's already installed
+if (hash protoc 2>/dev/null) ; then
+    echo "found systemwide protoc, not installing locally"
+else
+    echo "no systemwide protoc installation found"
+    echo "moving bin/protoc to ./devtools/bin/protoc"
+    mv protoc3/bin/* devtools/bin/
+fi
```

Comment:
Not sure I agree with this change. We may want a version we control, even if it is installed globally.

Reply:
I think it depends whether you feel that running a systemwide `protoc` is an error. I understand your point, that we may want to require a version we control, because a different version can result in different output that won't pass CI. However, this runs into two issues.

The first is FreeBSD, where at least the way it was set up in the PR we just merged, it relies on a systemwide installation: https://github.com/spacemeshos/go-spacemesh/pull/2058/files#diff-8269cac45755f62b44b6f8ab4426f294

The second is the upgraded Docker image in this PR. I'm actually not sure why this worked previously--maybe something changed in the more recent Docker images--but the binary protoc releases cannot be run on Alpine Linux because they're built using nonstandard musl libc. For this reason I modified the Dockerfile to instead install the systemwide package: https://github.com/spacemeshos/go-spacemesh/pull/2052/files#diff-3254677a7917c6c01f55212f86c57fbf

In any case, all of this will be going away in a matter of days when we merge the new API code, since we'll finally be able to remove all protobuf support, so I think there's no point in debating this!